aten/src/ATen/native/cuda/GridSampler.cuh

```diff
@@ -142,6 +142,19 @@ scalar_t reflect_coordinates_set_grad(scalar_t in, int twice_low, int twice_high
   }
 }
 
+template<typename scalar_t> 
+static __forceinline__ __device__ 
+scalar_t safe_downgrade_to_int_range(scalar_t x){
+  // -100.0 does not have special meaning. This is just to make sure 
+  // it's not within_bounds_2d or within_bounds_3d, and does not cause 
+  // undefined behavior. See #35506.  
+  if (std::is_same<at::Half, scalar_t>::value && !::isfinite(static_cast<double>(x)))
```

Comment:
why do you need a separate conditional here? half inf will still be picked up by the later conditional.

Reply:
I forgot to delete that. Thanks for pointing it out.